core/dbt/adapters/base/query_headers.py

```diff
@@ -87,18 +77,26 @@ def __init__(self, config: AdapterRequiredConfig, manifest: Manifest):
         self.comment = _QueryComment(None)
         self.reset()
 
-    def _get_comment_macro(self):
-        if (
-            self.config.query_comment != NoValue() and
-            self.config.query_comment
-        ):
+    def _get_comment_macro(self) -> Optional[str]:
+        default_comment = QueryComment().comment
+        if not self.config.query_comment:
+            return default_comment
+
+        if isinstance(self.config.query_comment, str):
+            if self.config.query_comment in ('None', ''):
+                # query comment is disabled.
+                return None
```

Comment:
I don't think the query comment should be `'None'`, so this should all just get handled by the if-block above (because if it's `None` or `''`, it should be the default comment). If it can be `'None'`, we should fix _that_ issue...